feat(err): show stacktrace as text

hpouillot opened the pull request:
## Problem

- selecting text closes the issue card and / or frame collapsible

## Changes

- Introduce a 'show as text' option
- Add events to test stacktrace rendering
- Add high-level properties on issue card
- Add copy filename button on frames

<img width="1404" alt="Screenshot 2025-04-23 at 16 05 52" src="https://github.com/user-attachments/assets/ab0d1dab-df33-48d4-b0ea-70208d5c922b" />

<img width="1186" alt="image" src="https://github.com/user-attachments/assets/e48006a1-27f9-4cf5-bae3-ac4de033bfa5" />

## How did you test this code?

using storybook

hpouillot committed:
add show as text display option

oliverb123 commented:
Imo you don't need the "show as text" button, I'd make it "copy as text" and auto-fill the clipboard (guessing it's for feeding into LLMs?), or remove it entirely.

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

Update UI snapshots for `chromium` (3)

Update UI snapshots for `chromium` (9)

Update UI snapshots for `chromium` (7)

Update UI snapshots for `chromium` (8)

Update frontend/src/scenes/error-tracking/components/Stacktrace/StacktraceBase.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

Update UI snapshots for `chromium` (5)

add copy filenames buttons

fix empty classnames

Update frontend/src/scenes/error-tracking/__mocks__/events/javascript_unresolved.json

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

Update frontend/src/scenes/error-tracking/components/ToggleButton/ToggleButton.stories.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

fix storybooks

fix tests

anonymize events

hpouillot commented:
That what I wanted to do at first, but users might want to copy just the file names => https://posthog.slack.com/archives/C07AA937K9A/p1744818436814289?thread_ts=1744802457.574199&cid=C07AA937K9A

oliverb123 commented:
Ah I see yeah - could we just put a little clipboard icon beside the file name on each frame instead?

daibhin commented:
Could we have both? "Copy" and "View as text" options

hpouillot commented:
Let's do that I'll add "copy filename" buttons. In any case, we need a top action to either "copy everything as text" of "show everything as text".

hpouillot committed:
fir storybook runner

hpouillot committed:
Merge branch 'master' into err/show-as-text

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

daibhin commented on frontend/src/lib/lemon-ui/icons/icons.tsx:
```diff
@@ -1513,3 +1513,14 @@ export function IconSlash(props: LemonIconProps): JSX.Element {
     )
 }
+
+export function IconFire(props: LemonIconProps): JSX.Element {
```
I'm not sure what the latest on icons is but we used to use this repo as a 3rd party dependency: https://github.com/posthog/icons

hpouillot replied:
Hum, ok good to know, I'll reuse the bug icon otherwise, works well here

daibhin commented on frontend/src/scenes/error-tracking/components/Stacktrace/StacktraceTextDisplay.stories.tsx:
```diff
@@ -0,0 +1,60 @@
+
+export default meta
+
+// Text stacktrace
+// export function TextDisplayFullLoading(): JSX.Element {
+//     const props = defaultBaseProps(
+//         'python_resolved',
+//         {
+//             loading: true,
+//         },
+//         true
+//     )
+//     return <StacktraceTextDisplay {...props} />
+// }
```
🔪

hpouillot replied:
I should probably add a comment on this one, I commented out because the CI relies on LemonSkeleton to know if the UI is loaded before taking a screenshot. I'll see if I can make it passes the CI.

daibhin commented on frontend/src/scenes/error-tracking/components/ContextDisplay.tsx:
```diff
@@ -82,3 +93,14 @@ function ContextRow({ label, value }: ContextRowProps): JSX.Element {
     )
 }
+
+function concatValues(attrs: ExceptionAttributes | null, ...keys: (keyof ExceptionAttributes)[]): string | undefined {
```
We have this function in multiple places

daibhin commented on frontend/src/scenes/error-tracking/components/ExceptionAttributes/ExceptionAttributesIconList.tsx:
```diff
@@ -0,0 +1,45 @@
+    }
+    return (
+        <LemonTag>
+            {cloneElement(children, { ...children.props, className: 'text-sm text-secondary' })}{' '}
```
Why do we need to clone the children here? Can we not add classNames to the `LemonTag` or do something like `<PropertyIcon property="$os" value={attributes.os} className="text-sm text-secondary" />` above?

daibhin commented on frontend/src/scenes/error-tracking/components/ToggleButton/ToggleButton.tsx:
```diff
@@ -0,0 +1,39 @@
+import { cva } from 'cva'
+import { ButtonPrimitive } from 'lib/ui/Button/ButtonPrimitives'
```
Any reason to use the button primitives here? Afaik we only use them in the new project side bar and I'm not they're as fleshed out as our regular buttons

hpouillot replied:
Yes those toggles are encapsulated in a button group and it's not supported by lemon buttons

daibhin commented on frontend/__snapshots__/errortracking-stacktracegenericdisplay--generic-display-empty--dark.png:
What are all these mini stories? Feels like we should get rid of them

daibhin commented on frontend/src/scenes/error-tracking/components/IssueCard.tsx:
```diff
@@ -1,61 +1,112 @@
+                    >
+                        <IconDocument />
+                    </ToggleButtonPrimitive>
+                    <ToggleButtonPrimitive
+                        iconOnly
+                        checked={showAllFrames}
+                        onCheckedChange={setShowAllFrames}
+                        tooltip="Show vendor frames"
+                    >
+                        <IconBox />
+                    </ToggleButtonPrimitive>
```
Clicking this does nothing for me

![frames](https://github.com/user-attachments/assets/c04b8c04-d7fd-40aa-947a-db8c4ca1c6e0)

hpouillot replied:
Yes, there is no "vendor" frames on this exception, button state is not clear enough I think

hpouillot replied:
<img width="211" alt="image" src="https://github.com/user-attachments/assets/da346c86-c7cf-4527-a0dd-6b61f2ebe166" />

daibhin commented on frontend/src/scenes/error-tracking/components/IssueCard.tsx:
```diff
@@ -1,61 +1,112 @@
-                    className="select-none"
-                />
+            <IssueCardActions className="absolute top-2 right-3 flex gap-2 items-center z-10">
+                <ButtonGroupPrimitive size="sm">
```
I'm a little bit weary about using the button primitives because they are not used outside of the project side bar
![buttons](https://github.com/user-attachments/assets/406b6c09-b2d8-44fb-bbe6-06c001dad675)


Any reason we can't use the designed buttons that replay uses from `frontend/src/lib/components/PanelLayout/PanelLayout.tsx`
![buttons-new](https://github.com/user-attachments/assets/df0cbedb-562f-498f-a243-1e5b866ba473)

hpouillot replied:
I like the structure of the button primitives (which relies on radix for accessibility) but it might be a bit early for now, I'll see if we can switch.

hpouillot replied:
I checked and it won't work with them, those are regular buttons and we need a button group here (rounded on the sides only)

daibhin commented on frontend/src/scenes/error-tracking/components/IssueCard.tsx:
```diff
@@ -76,42 +127,79 @@ export function IssueCard(): JSX.Element {
     )
 }
+
+// Use issue properties as a placeholder to real event properties
+// and abstract showing as text or generic
```
Not sure we need this, think it's pretty clear what is going on
```suggestion
```

hpouillot committed:
Update frontend/src/scenes/error-tracking/components/IssueCard.tsx

Co-authored-by: David Newell <d.newell1@outlook.com>

hpouillot committed:
Merge branch 'master' into err/show-as-text

hpouillot committed:
do not use cache for building

hpouillot committed:
Merge branch 'hp/fix-hogvm-build' into err/show-as-text

hpouillot committed:
remove duplicated concatValues

hpouillot committed:
do not clone children

hpouillot committed:
regroup stories

hpouillot committed:
remove empty snapshot

hpouillot committed:
fix stacktrace display

hpouillot committed:
make toggle state more visible

hpouillot committed:
remove icon fire

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)
